src/main/java/frc/robot/SwerveModule.java

```diff
@@ -120,20 +115,22 @@ public void setDesiredState(SwerveModuleState desiredState) {
         desiredState = SwerveModuleState.optimize(desiredState, getState().angle);
 
         // PID Controller for both translation and rotation
-        _transMotor.set(desiredState.speedMetersPerSecond / SwerveConstants.kPhysicalMaxSpeedMPS);
-        desiredRadians = desiredState.angle.getRadians();
-        PIDOutput = _rotPID.calculate(getRotPosition(), desiredState.angle.getRadians());
+        mTransMotor.set(desiredState.speedMetersPerSecond / SwerveConstants.PHYSICAL_MAX_SPEED_MPS);
+        mDesiredRadians = desiredState.angle.getRadians();
```

Comment:
Using instance variable mDesiredRadians like this is buggy. For some desiredState values, the setDesiredState function returns at line 111 without updating mDesiredRadians. In the event mDesiredRadians keeps stalled value, some logic, e.g. periodic(), logic in SwerveSubsystem class calls getDesiredRadiansRot() at 268 and getting the stalled mDesiredRadians.

Reply:
Since the call in the SwerveSubsystem is using the getDesiredRadians() method solely for debugging, keeping the optimization calculations after the return statement will help reduce computing overhead when the desired speed is less than the threshold.

Reply:
The debugging printout is wrong in that case then. Also there is no guarantee that getDesiredRadians() will not be used for other purposes in the future because it is public and there is no compilation or run-time guard against it.  

- Either add a new line `mDesiredRadians = 0` before returning in the middle of the function.
- Or add a guard logic so that  getDesiredRadians() only works in debug. In C++ people use conditional inclusion
 
`
#ifdef __DEBUG
  public double getDesiredRadiansRot() {
        return mDesiredRadians;
    }
#endif
`
In Java maybe use CompConstants.DEBUG_MODE and a checked exception or generics/template class.